xbmc/platform/win32/WinRtUtil.cpp

```diff
@@ -0,0 +1,203 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "WinRtUtil.h"
+
+#include "rendering/dx/DirectXHelper.h"
+#include "utils/SystemInfo.h"
+#include "utils/log.h"
+
+#ifdef TARGET_WINDOWS_DESKTOP
+#include <Windows.Graphics.Display.Interop.h>
+
+#else
+#include <winrt/Windows.Devices.Display.Core.h>
+
+using namespace winrt::Windows::Devices::Display::Core;
+#endif
+
+#include <winrt/Windows.Graphics.Display.h>
+
+using namespace winrt::Windows::Graphics::Display;
+
+#ifdef TARGET_WINDOWS_DESKTOP
+extern HWND g_hWnd;
+#endif
+
+static constexpr std::string_view AdvancedColorKindToString(AdvancedColorKind kind)
+{
+  switch (kind)
+  {
+    case AdvancedColorKind::StandardDynamicRange:
+      return "SDR";
+    case AdvancedColorKind::WideColorGamut:
+      return "WCG";
+    case AdvancedColorKind::HighDynamicRange:
+      return "HDR";
+    default:
+      return "unknown";
+  }
+}
+
+static void LogAdvancedColorInfo(AdvancedColorInfo colorInfo)
+{
+  std::string availableAdvColorKind;
+  std::array<AdvancedColorKind, 3> kinds{AdvancedColorKind::StandardDynamicRange,
+                                         AdvancedColorKind::WideColorGamut,
+                                         AdvancedColorKind::HighDynamicRange};
+
+  for (auto kind : kinds)
```

Comment:
```suggestion
  for (const auto& kind : kinds)
```

Reply:
ok for the principle. It's an enum so no performance gained,